src/components/PhotoSwipe/InfoDialog/index.tsx

```diff
@@ -58,12 +58,28 @@ export function FileInfo({
     }, [metadata]);
 
     useEffect(() => {
-        if (!location && exif) {
-            const exifLocation = getEXIFLocation(exif);
-            if (exifLocation.latitude || exifLocation.latitude === 0) {
-                setLocation(exifLocation);
+        const main = async () => {
+            if (!location && exif) {
+                const exifLocation = getEXIFLocation(exif);
+                if (exifLocation.latitude || exifLocation.latitude === 0) {
+                    setLocation(exifLocation);
+                    try {
+                        let updatedFile = await updateFileLocation(
```

Comment:
Should we also verify the ownership of the file? On Mobile, we let user view metadata for shared files.

Reply:
Oh ya, nice catch! I have added the check to verify it is part of shared collection and prevent update call in that case